File: test/sample.cs
// Sample C# file for testing code extraction
using System;
using System.Collections.Generic;

namespace CalculatorApp
{
    public class Calculator
    {
        private List<string> operations;

        public Calculator()
        {
            operations = new List<string>();
        }

        public int Add(int a, int b)
        {
            int result = a + b;
            string operation = $"{a} + {b} = {result}";
            operations.Add(operation);
            return result;
        }

        public int Multiply(int a, int b)
        {
            int result = a * b;
            string operation = $"{a} * {b} = {result}";
            operations.Add(operation);
            return result;
        }

        public string GetHistory()
        {
            return string.Join("\n", operations);
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            Calculator calc = new Calculator();
            calc.Add(5, 3);
            calc.Multiply(4, 2);
            Console.WriteLine(calc.GetHistory());
        }
    }
}

File: test/sample.rs
// Sample Rust file for testing code extraction
use std::collections::VecDeque;

struct Calculator {
    operations: Vec<String>,
}

impl Calculator {
    fn new() -> Self {
        Calculator {
            operations: Vec::new(),
        }
    }

    fn add(&mut self, a: i32, b: i32) -> i32 {
        let result = a + b;
        let operation = format!("{} + {} = {}", a, b, result);
        self.operations.push(operation);
        result
    }

    fn multiply(&mut self, a: i32, b: i32) -> i32 {
        let result = a * b;
        let operation = format!("{} * {} = {}", a, b, result);
        self.operations.push(operation);
        result
    }

    fn get_history(&self) -> String {
        self.operations.join("\n")
    }
}

fn main() {
    let mut calc = Calculator::new();
    calc.add(5, 3);
    calc.multiply(4, 2);
    println!("{}", calc.get_history());
}